### PR TITLE
feat: markdown support in report's description

### LINKED ISSUE #6527: Allow to use markdown on report description
### NetBox version

v2.11.3

### Feature type

Change to existing functionality

### Proposed functionality

I'd like to propose an enhancement on the look & feel of the r[eport_list.html](https://github.com/netbox-community/netbox/blob/develop/netbox/templates/extras/report_list.html) page.

Each report is currently presented in a List of 4 columns (_Name_ | _Status_ | _Description_ | _Last Run_). I would propose to render the _Description_ column using markdown support.

The change would be as simple as modifying the report_list.html template and add the class `rendered-markdown` and use the `markdown` django template function on the following line

https://github.com/netbox-community/netbox/blob/8d0a3c8e6932f4c523919180d3e75608cdf54a9b/netbox/templates/extras/report_list.html#L32

The markdown support has already been implemented for each report log messages so it won't be difficult to use the same feature for the Report description (check this [commit](https://github.com/netbox-community/netbox/commit/7c48e3632db052924c80213fc07faa8d3059fe7c))

I can handle this issue and propose a Pull Request.

### Use case

Adding markdown support to each Report description allows to structure the this description easily if we want to explain in more detail the whole test case covered.

### Database changes

None

### External dependencies

None

### PR DESCRIPTION
**Pull request by maximumG**
### Fixes: #6527

* markdown support in report list and report result pages
* Add notes in the documentation regarding markdown

### CODE REVIEW

File: docs/additional-features/reports.md [hunk 1]
```diff
@@ -85,6 +85,9 @@ As you can see, reports are completely customizable. Validation logic can be as
 !!! warning
     Reports should never alter data: If you find yourself using the `create()`, `save()`, `update()`, or `delete()` methods on objects within reports, stop and re-evaluate what you're trying to accomplish. Note that there are no safeguards against the accidental alteration or destruction of data.
 
+!!! note
```
Comment (jeremystretch) [inline docs/additional-features/reports.md:88]: We don't need call-outs specifically for this; just a note within the text mentioning Markdown support is sufficient.

File: docs/additional-features/reports.md [hunk 2]
```diff
@@ -95,6 +98,9 @@ The following methods are available to log results within a report:
 
 The recording of one or more failure messages will automatically flag a report as failed. It is advised to log a success for each object that is evaluated so that the results will reflect how many objects are being reported on. (The inclusion of a log message is optional for successes.) Messages recorded with `log()` will appear in a report's results but are not associated with a particular object or status.
 
+!!! note
```
Comment (jeremystretch) [inline docs/additional-features/reports.md:101]: Same as above

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Thanks for this!

File: docs/additional-features/reports.md (hunk 1 above)
Comment (maximumG) [reply docs/additional-features/reports.md]: Removed in commit 10c9954

File: docs/additional-features/reports.md (hunk 2 above)
Comment (maximumG) [reply docs/additional-features/reports.md]: Removed in commit 10c9954

**Review by jeremystretch — APPROVED** [review]

**Pull request merged**